### ISSUE TITLE
Unknown movie. Movie found does not match wanted movie.

### ISSUE THREAD

**Issue by vendeka** [Radarr/Radarr#1407]
I am using PTP with Radarr and some releases have different or wrong naming (e.g. typo) but those releases shouldn't be refused by Radarr as PTP supports IMDB ID searching so the results based on IMDB ID search are always correct.

I would like to provide a movie which I recently have problem:

Prelude to War (1942)
http://www.imdb.com/title/tt0035209/

Radarr Ver. 0.2.0.596

![image](https://cloud.githubusercontent.com/assets/3618229/25105451/4edbf54c-23cd-11e7-85ca-9eb968578e4c.png)

**Comment by galli-leo** [comment]
This should be happening. not sure why. The second one though is due to the year being wrong.

**Comment by Twanislas** [comment]
Hi !

I face the same issue, could it be related to the accents or language ?

For both the error (hovering on the "!") says : "Unknown movie. Movie found does not match wanted movie."

Let me know how/if I can help :)

![radarr_bug_match](https://cloud.githubusercontent.com/assets/524706/25144121/0f97b2e0-246d-11e7-97eb-ecfa02755376.png)
![radarr_bug_match2](https://cloud.githubusercontent.com/assets/524706/25144123/0faa5a80-246d-11e7-8523-f2920aee356b.png)

**Comment by galli-leo** [comment]
@Twanislas Nope. The problem here is that the year is at the end of the filename rather than right after the movie title.

**Comment by Twanislas** [comment]
Thanks for the fast response ! :)

Is there a way to "force" Radarr to use a specific release ? Like "Dude, use this one, I tell ya, that's the right one" :D

It would then override everything and download + process it ?

Cheers !

**Comment by vendeka** [comment]
Since PTP search results are filtered by IMDB ID, I think it is safe to ignore extra checks like year, exact name etc. At least, there should be an option to ignore such extra checks.

**Comment by Firemad03** [comment]
I got one for you. 

![image](https://cloud.githubusercontent.com/assets/24792889/25208711/5bff9b34-2544-11e7-956b-96088ebe0e31.png)

The movies doesn't have the year in the title but it does in the filename.

**Comment by Firemad03** [comment]
Also I can't manually add it because of the error. We need to be able to download it if we manually search for it. So I'm stuck for this movie ;)

**Comment by galli-leo** [comment]
@Firemad03 The problem is, even if you did manually download it through Radarr, it won't be able to import it, since the movie does not match. We will have to come up with a better system.

**Comment by Quiksmage** [comment]
+1 on what @vendeka said.

**Comment by jooncheez** [comment]
I got another example here. It seems to be the accents though?

![image](https://cloud.githubusercontent.com/assets/10760947/25400828/1ab1191e-29f4-11e7-867b-c1f8c83facec.png)

**Comment by galli-leo** [comment]
@jooncheez Can you open a new issue for that one? I know what's causing it.

**Comment by jooncheez** [comment]
@galli-leo Sure thing: https://github.com/Radarr/Radarr/issues/1441

**Comment by Wohlraj** [comment]
Hi,

Stumbled upon this issue.
Example here: English title is Hana yori mo naho (from IMDB), shows up as Hana on Radarr.
![screenshot from 2017-05-01 15-01-30](https://cloud.githubusercontent.com/assets/6720382/25580228/804e1260-2e7f-11e7-9f0a-56de851f4846.png)

**Comment by silv3r23** [comment]
hello, german movies with tags which are not following the scene rules (order) are not matching
Radarr Version: 0.2.0.654
![2017-05-02 22_16_57-der soldat james ryan - radarr](https://cloud.githubusercontent.com/assets/28316411/25637924/e33d90a0-2f86-11e7-8c1c-b0a978645fd2.png)
[1james.txt](https://github.com/Radarr/Radarr/files/971902/1james.txt)

**Comment by galli-leo** [comment]
@silv3r23 Maybe tell them to adhere to scene rules? Not much we can do here, since we don't want e.g. The Danish Girl or Speaking German to be matched wrongly.

**Comment by Miloune** [comment]
Hello,

Same problem with this if it can help you:
![image](https://cloud.githubusercontent.com/assets/8145119/25680686/f6f240fc-3051-11e7-9751-a89678ca0651.png)

**Comment by galli-leo** [comment]
@Miloune Same problem: Not adhering to scene roles. Year needs to immediately come after the title.

**Comment by werkkrew** [comment]
A couple more examples:

![example1](https://cloud.githubusercontent.com/assets/838837/25688293/d1a00600-304b-11e7-9a8c-b6b5b2409ac7.png)

![example2](https://cloud.githubusercontent.com/assets/838837/25688295/d6e683d2-304b-11e7-81af-fabd2ff2ba47.png)

**Comment by silv3r23** [comment]
@galli-leo 
> Year needs to immediately come after the title.
I think it is necessary that we came up with another method of parsing. The [year] tag is optional in some [cases](https://scenerules.org/t.html?id=2013_DE_SDX264.nfo)

**Comment by galli-leo** [comment]
@silv3r23 The problem here is that we do not know what when the movie title stops / when the other stuff starts without the year. We kinda use the year as an anchor to determine things. Additionally, you really want the year or else you are gonna get movies downloaded you don't want, e.g. Beauty and the Beast (2017 vs 2016 vs 19something)
AFAIK 98% of releases are fine, and the other ones just need to learn to stick to scene rules. Sonarr also needs e.g. season or year for it to parse episodes.

**Comment by galli-leo** [comment]
@werkkrew Fanboys seems the TMDB date is wrong (In cinemas February 2009). Maybe correct that on TMDB itself? The other one I don't really know what could be causing that. I will have to check it out.

**Comment by galli-leo** [comment]
@silv3r23 Interesting read the thing you linked. I think that's really stupid as Radarr cannot know whether German is part of the title or part of the language tag. IMO they should change it to scene standards.

**Comment by werkkrew** [comment]
@galli-leo 

Actually, it seems that the scene might be wrong in this case.  IMDB, Wikipedia, and TMDB all agree on the release being in 2009 but every single scene release says 2008.  Interestingly RT also uses 2008.  

I am not familiar enough with the scene naming rules to know exactly what date they use on a release.

I am also not sure how to rectify a situation like this other than to allow a manual override in Radarr or something.

Oldboy is a similiar issue, when I search for oldboy within radar the matching result is "Oldboy (2002)" but every other resource seems to use 2003 as the date of release.

![example3](https://cloud.githubusercontent.com/assets/838837/25720932/a3dfb526-30dc-11e7-81aa-1fd92c131361.png)

**Comment by florian-lemiere** [comment]
Hello everyone!!! And first of all thank you to the developers for the work.

I encounter the same problem with the french torrent9 provider. All titles do not comply with the scheme and are therefore impossible to download. Is it possible to override this problem?

**Comment by Twanislas** [comment]
+1 for Torrent9
And +1 for the thanks ! ;)

**Comment by galli-leo** [comment]
@werkkrew Scene should be using imdb date. So indeed they are wrong :)
@florian-lemiere @Twanislas I do have some plans for overwriting the movie matching decision, but that will have to come at a later date. I could enable downloading for those movies, though Radarr wouldn't be able to import them then.

**Comment by florian-lemiere** [comment]
Ok. Since the scheme used by Torrent 9 to name all its files seems to be always the same, it might be possible to integrate a feature that would cut the filename into a regular expression, match it to the schema requested, and would rename?
Let me know if I can help

**Comment by galli-leo** [comment]
@florian-lemiere We are already using regular expressions for matching movie titles. The problem is that if we do not have the year immediately after the movie title, we cannot know (though any regular expressions) what is in the movie title and what is just additional tags. An example might be "XXX: Extended 2017". Is Extended there for extended edition? Does it belong to a movie title? We cannot possibly know. Another example: "Stop it French 2016". Is this movie in French or is French part of the title? Here we cannot know that too. So either Torrent9 has to change their naming scheme (which they should) or somehow Jackett (Or whatever you are using) would have to extract either the IMDBid from Torrent9 or the correct release title (Which I have seen on torrent9).

**Comment by silv3r23** [comment]
@galli-leo I think I have a solution for all these movie parsing problem. almost finished. how can I contact you or someone of the developer team?

**Comment by florian-lemiere** [comment]
@galli-leo 
Okay. Thank you for that clear answer. It is true that I had not thought of these particular cases. I will look for my part in the configuration of Torrent 9 for Jackett. Also, maybe the silv3r23 solution will be the right one? To follow! ... Thank you and again congratulations for the work.

**Comment by galli-leo** [comment]
@vendeka Sorry for the late response. But we wont probably be removing the year check from PTP, since it's at a much lower level than PTP (making it difficult to separate those two). Additionally, before we had the year check with imdbid matching, often releases from indexers returned a wrong imdbid. For example, Beauty and the Beast 2013 would return the imdbid of Beauty and the Beast 2017, hence why we started including the year check.

**Comment by galli-leo** [comment]
@silv3r23 Either hop on discord, open a new issue detailing what you would suggest or if you are already integrating it into Radarr, create a Pull Request.

**Comment by chuckkay** [comment]
![zodiac](https://cloud.githubusercontent.com/assets/10617746/25862317/9aeffd54-34b5-11e7-9d5b-3b192003c8cb.JPG)

**Comment by Quiksmage** [comment]
hover over the red !. it will tell you why you can't download it. Or at least, why it thinks it shouldn't.

**Comment by galli-leo** [comment]
@chuckkay **In Cinemas: December 2016**. TMDB apparently has the wrong year there. I would suggest you go ahead and correct it there, so all of us can benefit :) But there is nothing much we can do in this situation.

**Comment by Voteax** [comment]
I have one with matched Year Dates:
![2017-05-10 18_34_15-der herr der ringe - die ruckkehr des konigs - radarr](https://cloud.githubusercontent.com/assets/12576956/25910099/82677d8e-35af-11e7-8be1-ab73e4e80ef2.png)

Maybe because of the "ü" and "ö" in the Movie Title?
Only a DVD Release is found. I don´t know why.

**Comment by galli-leo** [comment]
@Voteax Actually the ö's should be correctly replaced with oe, so not sure why they do not match.

**Comment by JDIacobbo** [comment]
I would like to add my case to the thread. It looks like the name and the year match and there are not special characters involved.

![screenshot 2017-05-15 at 110939 pm](https://cloud.githubusercontent.com/assets/7070260/26088347/b08bc896-39c4-11e7-9cd1-e4b84025ba5c.jpg)

**Comment by Ninjouz** [comment]
Hello @galli-leo,

Here's another one (automatic search during the night via IMDB/TMDB usenet indexer) : 

```
Debug|Parser|Movie Parsed. blow-logan - 2017 
Debug|Parser|Language parsed: English
Debug|QualityParser|Trying to parse quality for blow-logan.2017.1080p.bluray.x264
Debug|Parser|Quality parsed: Bluray-1080p v1
Debug|Parser|Release Group parsed: logan
Debug|Parser|Parsing string 'blow-logan.2017.1080p.bluray.x264'
Unknown Movie blow-logan.2017.1080p.bluray.x264
```

**Comment by Quiksmage** [comment]
@Ninjouz , That's going to be because the releaser's tag is at the beginning of the file. That isn't natural. I ended up having to manually import that release too. It should have trailed with -BLOW

**Comment by hed0nist** [comment]
Hello.
I posted a fix for torrent9 in github/jackett. The goal is to send correct data according scene rules.
I dont know github environnement and sending pull requests.
Here is the link : https://github.com/Jackett/Jackett/issues/1410

Here is the modified code for torrent9 ( far from perfect ) but it could be a good beginning for people to change a few lines and get things working with another torrents site which dont use Title.Date.Lang.Format as it is requested.

> 
fields:
  site_date:
    selector: td:nth-child(1) a
    filters:
      # date is at the end of the title, so we get it and name it site_date
      - name: regexp
        args: "(\\w+)$"
  title:
    selector: td:nth-child(1) a
    filters:
      # now we put the date at the right place according scene naming rules using .Result.site_date
      - name: replace
        args: ["FRENCH", "{{ .Result.site_date }} FRENCH"]
      - name: replace
        args: ["TRUEFRENCH", "{{ .Result.site_date }} TRUEFRENCH"]
      - name: replace
        args: ["VOSTFR", "{{ .Result.site_date }} VOSTFR"] 
      # and we delete it at the end
      - name: re_replace
        args: ["(\\w+)$", ""]

**Comment by Ninjouz** [comment]
@Quiksmage I know that but still the regexp should be able to detect that (guessit python lib is able to).

**Comment by jeramie72** [comment]
![image](https://cloud.githubusercontent.com/assets/18276746/26481871/b3aa7c08-4198-11e7-9b53-e164544f572f.png)

looks like the date is different?
would it not be possible to still show the manual download link and then allow to manually import the title into the radarr database? hope that made sense.

**Comment by galli-leo** [comment]
@jeramie72 Yeah, I want to add a system where you can force add such torrents and Radarr will remember the error (e.g. wrong year) so it can be auto imported. I might even add a way to share these things so other people can download them then correctly or something.

**Comment by wolffstarr** [comment]
Another one here. I'm guessing this is at least partly because of the fact that the subtitle (not subs) is listed as part of the title in everywhere except the releases. I could see this being a thorn in the side, but maybe an option to match on titles prior to colons might work? Anyhow, in this case the first half of the title and the year all match up.

![unknown-release-movie](https://cloud.githubusercontent.com/assets/4945910/26516449/85866c9e-4254-11e7-84cf-092a4c7472f5.png)

**Comment by galli-leo** [comment]
@wolffstarr I don't think that will work. However, I suggest adding the title (without the colon) as an alternative title on themoviedb.org. This will also help out future users.

**Comment by silv3r23** [comment]
hey guys! I optimized the parser (logging too) and for me Radarr works much better! I would be glad if a few people would test it that I can commit my changes in a few days.
1. set the debug mode from Debug to Trace in webinterface (Setting > General > Logging)
2. stop service
2. replace NzbDrone.Core.dll and NzbDrone.Core.pdb
3. start service
[http://www.xup.in/dl,36376233/NzbDrone.Core.zip/](url)

if you have still problems please post your radarr.trace.txt within the movie (STRG+F) which is not working!
Note: currently matching for releases without the year-tag wont work. like some actual releases from PsO or DETAiLS. I hope I will find a solution for this!

**Comment by galli-leo** [comment]
@silv3r23 Could you open a pr for that? Not feeling comfortable with people downloading random dlls from the internet and replacing them :/

**Comment by silv3r23** [comment]
@galli-leo sure, just done =)

**Comment by silv3r23** [comment]
bug fixes:
[https://drive.google.com/open?id=0BwmvoOUPL1lkUEEyX1htcGNOV1U](NzbDrone.Core.dll)

**Comment by ToRvaLDz** [comment]
Here is my log, same problem

```
17-6-12 18:19:03.5|Debug|Parser|Parsing string 'Mission Impossible: Rogue Nation (2015)�[XviD - Ita Ac3 - SoftSub Ita]azione, spionaggio, thriller *Prima Visione* Team mulnic Tom Cruise'
17-6-12 18:19:03.5|Trace|Parser|^(?<title>(?![(\[]).+?)?(?:(?:[-_\W](?<![)\[!]))*\(?(?<edition>(((Extended.|Ultimate.)?(Director.?s|Collector.?s|Theatrical|Ultimate|Final(?=(.(Cut|Edition|Version)))|Extended|Rogue|Special|Despecialized|\d{2,3}(th)?.Anniversary)(.(Cut|Edition|Version))?(.(Extended|Uncensored|Remastered|Unrated|Uncut|IMAX|Fan.?Edit))?|((Uncensored|Remastered|Unrated|Uncut|IMAX|Fan.?Edit|Edition|Restored|((2|3|4)in1))))))\)?.+(?<year>(19|20)\d{2}(?!p|i|\d+|\]|\W\d+)))+(\W+|_|$)(?!\\)
17-6-12 18:19:03.5|Debug|Api|[GET] /api/log?page=1&pageSize=50&sortKey=time&sortDir=desc&filterType=equal: 200.OK (189 ms)
17-6-12 18:19:03.5|Debug|Parser|Movie Parsed. Mission Impossible - 2015 
17-6-12 18:19:03.6|Debug|Parser|Language parsed: Italian
17-6-12 18:19:03.6|Debug|QualityParser|Trying to parse quality for Mission Impossible: Rogue Nation (2015)�[XviD - Ita Ac3 - SoftSub Ita]azione, spionaggio, thriller *Prima Visione* Team mulnic Tom Cruise
17-6-12 18:19:03.6|Debug|Parser|Quality parsed: Unknown v1
17-6-12 18:19:03.6|Debug|Parser|Release Group parsed: 
17-6-12 18:19:03.6|Debug|ParsingService|missionimpossible
17-6-12 18:19:03.7|Debug|ParsingService|missionimpossible
....
17-6-12 18:19:04.5|Debug|ParsingService|missionimpossible
17-6-12 18:19:04.5|Debug|ParsingService|No matching movie Mission Impossible
17-6-12 18:19:04.5|Debug|DownloadDecisionMaker|Release rejected for the following reasons: [Permanent] Unknown movie. Movie found does not match wanted movie.
```

**Comment by samjovanou** [comment]
Can i maybe suggest adding a column in the movie info next to the release date and also in the search results window relating to the imdbid or the like that is used to verify the release?
So at least if a search comes up we know whether or not that is the culprit? Forgive me if this has nothing to do with the search process but it is my understanding it has some bearing on it?
Im having an issue on a few movies which are named perfectly and all tags are used as per western scene rules but still comes as an unmatch.
In fact the search results coming up seem to be completely off for some reason; barely related to the title. Can screenshot examples if you like.

**Comment by galli-leo** [comment]
@samjovanou Yeah currently working on that in a separate branch. Then the search result should show what went wrong in detail. E.g. Expected year to be 2016, but found year 2017, etc.

**Comment by galli-leo** [comment]
@ToRvaLDz I have opened a separate issue for that. Currently fixed on a separate branch #1676

**Comment by galli-leo** [comment]
@Twanislas Currently have a fix ready that should solve the issue with those releases.

**Comment by galli-leo** [comment]
Also currently planning a few things for fixing a lot of issue in this thread:

1. Force downloading a movie with wrong year or title. Title or year then get's added to a database, so Radarr remembers that the title or year were actually correct.
2. A way to automatically share those corrections, so others can profit too.
3. (Shown above) Being much more descriptive why a movie was not considered to be the correct one.
4. Gather all issues arising from RSS Sync (So movie not mapped correctly, too small, too few seeders, etc.) in one place below the movie, so you can improve your profiles, filters, etc.
5. A way to make mapping much more lenient, so that only parts of the release name have to correspond to parts of the title Radarr has.

**Comment by samjovanou** [comment]
Thanks for your hard work you are a champion!

**Comment by galli-leo** [comment]
Number 6. is now implemented on the latest nightly! Number 5. will follow (currently working great on a separate branch, needs some more testing though)

**Comment by d4re** [comment]
Thanks for adding 6 already! Radarr added quite a few previously missing releases thanks to the new lenient option.

But there seems to be some small issue remaining. Radarr added "Kong.Skull.Island.WEBRiP.LD.GERMAN.x264-SOV{{EteQCAuu68}}" to the download client automatically but stopped tracking it after that. So it was downloaded but Radarr didn't import it.

Let me know if you can use some more info regarding this. I'm happy to help.

**Comment by galli-leo** [comment]
@d4re Could be that I did not yet update the download tracking code, will take a look when I find some more free time. What is the file named like in your download client?

**Comment by d4re** [comment]
In the download client, it's called "Kong Skull Island WEBRiP LD GERMAN x264-SOV".
So exactly the same except for spaces instead of dots and no password at the end.

I think you did already update the download tracking code, since e.g "Ghost in the Shell German DL AC3 Dubbed 1080p WebHD h264-PsO" and "Table 19 Liebe ist fehl am Platz German DL 720p BluRay x264-KiNOWELT" where tracked correctly.

**Comment by galli-leo** [comment]
Yeah, so the problem here is that German does not immediately follow the release title (Kong Skull Island). So 6. is actually useless here. Though 5. should work here.

**Comment by d4re** [comment]
Yes, that seems to be the difference between this release and the others. And I get that 6 is not supposed to help in this case. 
But it's a bit confusing, that Radarr did find it and added it to the download queue automatically, so there seems to be a difference between parsing during release search and parsing in the download client.

**Comment by galli-leo** [comment]
@d4re Your indexer probably gave us the imdbid, so we were able to map it. However, after that, we do not have the imdbid anymore, so we cannot map it.

**Comment by d4re** [comment]
@galli-leo okay, now it makes sense. I didn't know it uses the imdb id to add releases now. Thanks for the clarification.

**Comment by galli-leo** [comment]
Latest nightly now includes 2. & 5.!

**Comment by adamgoldberg** [comment]
May I suggest that title matching should be irrespective of punctuation?  In other words, "LA Confidential" should match "L.A. Confidential".  For matching purposes, non alphanumeric characters can be ignored...

**Comment by galli-leo** [comment]
@adamgoldberg We have been doing that since the beginning.

**Comment by adamgoldberg** [comment]
Um, nope.  "LA Confidential" didn't match "L.A. Confidential".

**Comment by galli-leo** [comment]
@adamgoldberg Actually, the problem is that, it removes the A. in L.A. from the title when comparing but not the A in LA. Can you open a separate issue for that one?

**Comment by silv3r23** [comment]
```
17-7-16 14:51:27.0|Debug|Parser|Parsing string 'Die.Mumie.WEBRip.LD.German.x264-PsO{{pass}}'
17-7-16 14:51:27.0|Trace|Parser|^(?<title>(?![(\[]).+?)((\W|_))(?:(?<!(19|20)\d{2}.)(German|French|TrueFrench))(.+?)(?=((19|20)\d{2}|$))(?<year>(19|20)\d{2}(?!p|i|\d+|\]|\W\d+))?(\W+|_|$)(?!\\)
17-7-16 14:51:27.1|Debug|Parser|Movie Parsed. Die Mumie WEBRip LD - 0 
17-7-16 14:51:27.1|Debug|Parser|Language parsed: German
```
v0.2.0.778
Parser Leniency: Lenient Mapping

**Comment by galli-leo** [comment]
@silv3r23 Yeah, this is a problem, because the German tag is in the wrong place. It should come after the title :(

**Comment by stairmast0r** [comment]
I'm having an issue with a movie being detected properly, due to the uploader putting the redux year in the name instead of the original release year. Is this an issue that's being worked on, or is there an existing solution that I'm missing? Thanks :)

![image](https://user-images.githubusercontent.com/5339003/29773487-bb6e20fc-8bcb-11e7-984a-6e3cb1482935.png)

**Comment by galli-leo** [comment]
@stairmast0r That should be eliminated with force downloading, coming in a few hours to develop. :)

**Comment by arontx** [comment]
@galli-leo Are you saying that item (1) "force downloading" will be coming soon? Got a few releases I've been holding off on grabbing

**Comment by galli-leo** [comment]
@arontx It's actually on the latest develop release.

**Comment by Quiksmage** [comment]
Currently on latest develop.
I assume this is due to the name of the movie being "It 2017", but
It.Comes.At.Night.2017.1080p.BluRay.x264.BLOW downloaded tonight mapping to
It 2017

**Comment by ltjdym** [comment]
wheres force downloading?

**Comment by galli-leo** [comment]
@Cliffield You have to enable lenient mapping.

**Comment by danz1995** [comment]
Hey, anything on this? It has been almost 1 year.

**Comment by timmish** [comment]
Would love an update as well, especially cause I switched to v3 recently and everything else works perfectly.
I'm downloading lots of german movies and this is the largest issue I have at the moment.
